framework/src/outputs/Exodus.C

```diff
@@ -531,3 +537,9 @@ Exodus::handleExodusIOMeshRenumbering()
     _mesh_contiguous_numbering = true;
   }
 }
+
+Real
+Exodus::getTimeStepForOutput()
```

Comment:
Maybe rename this with `getOutputTime`? Time step to me means an integer step number. This function represents a right direction in my opinion and should be used by all outputs. So I also suggest to move it to the base class and let all outputs stop calling `time()` directly.